api/pkg/apis/v1alpha1/providers/stage/materialize/materialize.go

```diff
@@ -187,32 +187,6 @@ func (i *MaterializeStageProvider) Process(ctx context.Context, mgrContext conte
 						return outputs, false, v1alpha2.NewCOAError(nil, fmt.Sprintf("Empty instance name: catalog - %s", name), v1alpha2.BadRequest)
 					}
 
-					instanceName := instanceState.ObjectMeta.Name
-					parts := strings.Split(instanceName, ":")
-					if len(parts) == 2 {
-						instanceState.Spec.RootResource = parts[0]
-						instanceState.Spec.Version = parts[1]
-					} else {
-						mLog.Errorf("Instance name is invalid: instance - %s, catalog - %s", instanceName, name)
-						return outputs, false, v1alpha2.NewCOAError(nil, fmt.Sprintf("Instance name is invalid: catalog - %s", name), v1alpha2.BadRequest)
-					}
-
-					mLog.Debugf("  P (Materialize Processor): check instance contains %v, namespace %s", instanceState.ObjectMeta.Name, namespace)
-					_, err := i.ApiClient.GetInstanceContainer(ctx, instanceState.Spec.RootResource, namespace, i.Config.User, i.Config.Password)
-					if err != nil && strings.Contains(err.Error(), constants.NotFound) {
-						mLog.Debugf("Instance container %s doesn't exist: %s", instanceState.Spec.RootResource, err.Error())
-						instanceContainerState := model.InstanceContainerState{ObjectMeta: model.ObjectMeta{Name: instanceState.Spec.RootResource, Namespace: namespace, Labels: instanceState.ObjectMeta.Labels}}
-						containerObjectData, _ := json.Marshal(instanceContainerState)
-						err = i.ApiClient.CreateInstanceContainer(ctx, instanceState.Spec.RootResource, containerObjectData, namespace, i.Config.User, i.Config.Password)
-						if err != nil {
-							mLog.Errorf("Failed to create instance container %s: %s", instanceState.Spec.RootResource, err.Error())
-							return outputs, false, err
-						}
-					} else if err != nil {
-						mLog.Errorf("Failed to get instance container %s: %s", instanceState.Spec.RootResource, err.Error())
-						return outputs, false, err
-					}
-
 					instanceState.ObjectMeta = updateObjectMeta(instanceState.ObjectMeta, inputs)
 					objectData, _ := json.Marshal(instanceState)
 					mLog.Debugf("  P (Materialize Processor): materialize instance %v to namespace %s", instanceState.ObjectMeta.Name, instanceState.ObjectMeta.Namespace)
```

Comment:
Where do you call create instance? I saw you keep createInstanceContainer, is this intended?